### PR TITLE
Automatically compute schema version of system tables

### PR DESCRIPTION
**Pull request by tgrabiec**
Schema of system tables is defined statically and table_schema_version needs to be explicitly set in code like this:

```
builder.with_version(system_keyspace::generate_schema_version(table_id, version_offset));
``` 

Whenever schema is changed, the schema version needs to change, otherwise we hit undefined behavior when trying to interpret mutation data created with the old schema using the new schema.

It's not obvious that one needs to do that and developers often forget to do that. There were several instances of mistakes of omission, some caught during review, some not, e.g.: 31ea74b96e115b7c469631b8e14d49e0e2d83980. 

This patch changes definitions to call the new `schema_builder::with_hash_version()`, which will make the schema builder compute version from schema definition so that changes of the schema will automatically change the version. This way we no longer rely on the developer to remember to bump the version offset.

All nodes should arrive at the same version, which is verified by existing `test_group0_schema_versioning` and a new unit test: `test_system_schema_version_is_stable`.

### CODE REVIEW

**Comment by kostja** [comment]
How does the version for the same content that is generate automatically compare with the version generated manually previously? Is it not going to be equal, is it? So if I have a heterogeneous cluster with the same system keyspace it will never converge, won't it?

**Comment by kostja** [comment]
@tgrabiec I never thought actually what happens in a mixed cluster during upgrade, but repeating schema merge requests can kill cluster performance. Clusters can be stuck in heterogeneous state for hours or even days. How does it work today?

**Comment by tgrabiec** [comment]
> @tgrabiec I never thought actually what happens in a mixed cluster during upgrade, but repeating schema merge requests can kill cluster performance. Clusters can be stuck in heterogeneous state for hours or even days. How does it work today?

System keyspace schema is not included in schema digest and we don't pull it. When we exchange mutations (e.g. raft commands) we send canonical_mutation(s) which are interpreted using local schema. 

```
future<table_schema_version> calculate_schema_digest(distributed<service::storage_proxy>& proxy, schema_features features)
{
    return calculate_schema_digest(proxy, features, std::not_fn(&is_system_keyspace));
}
```

File: utils/hashing.hh [hunk 1]
```diff
@@ -76,6 +76,22 @@ struct appending_hash<bool> {
...
 };
 
+template<>
+struct appending_hash<double> {
+    template<typename H>
+    requires Hasher<H>
+    void operator()(H& h, double d) const noexcept {
+        // Mimics serializer for CQL double type, for inter-machine stability.
+        if (std::isnan(d)) {
+            d = std::numeric_limits<double>::quiet_NaN();
```
Comment (avikivity) [inline utils/hashing.hh:86]: I hope this is stable across compilers and architectures.

**Pull request merged**